modules/nw-ovn-k-baseline-adminnetwork-policy.adoc

```diff
@@ -0,0 +1,137 @@
+
+.Example YAML file for BANP
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: priority-50-example <1>
+spec:
+  priority: 50 <2>
+  subject:
+    namespaces:
+      matchLabels:
+          kubernetes.io/metadata.name: gryffindor
+  ingress:<3>
+  - name: "deny-all-ingress-from-slytherin"
+    action: "Deny"
+    from:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+  egress:<3>
+  - name: "pass-all-egress-to-slytherin"
+    action: "Allow" <4>
+    to:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+----
+<1> Policy name
+<2> `spec.priority` supports a maximum of 100 BANP in the values of 0-99 in a cluster. The lower the value the higher precedence. Creating `BaselineAdminNetworkPolicy` with the same priority because the outcome is nondeterministic.
+<3> BANP have both ingress and egress rules.
+<4> BANP `action` rules for ingress and egress accepts values of `Deny` and `Allow`.
+====
+
+
+[discrete]
+[id="BaselineAdminNetworkPolicy-default-deny-example"_{context}]
+== BaselineAdminNetworkPolicy Deny example
+The following BANP singleton ensures that the administrator has set up a default deny policy for all ingress monitoring traffic coming into the tenants at `internal` security level. When combined with the "AdminNetworkPolicy Pass example", this deny policy acts as a guardrail policy for all ingress traffic that is passed by the ANP `pass-monitoring` policy.
+
+.Example YAML file for a guardrail Deny rule
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: BaselineAdminNetworkPolicy
+metadata:
+  name: default
+spec:
+  subject:
+    namespaces:
+      matchLabels:
+        security: internal
+  ingress:
+  - name: "deny-ingress-from-monitoring"
+    action: "Deny"
+    from:
+    - namespaces:
+        namespaceSelector:
+          matchLabels:
+# ...
+----
+====
+
+[discrete]
+[id="BaselineAdminNetworkPolicy-multi-tenancy-example"_{context}]
+== BaselineAdminNetworkPolicy multi-tenancy example
+
+The following example demonstrates how `AdminNetworkPolicy` with `Pass` action can be used by administrators in conjunction with `BaseleineNetworkPolicy to create a multi-tenancy policy that allows one tenant to collect monitoring data on their application while simultaneously not collecting data from the second tenant.
+
+As an administrator if you apply both the "AdminNetworkPolicy Pass action example" and the "BaselineNetwork Policy Deny example", tenants are then left with the ability to choose to create a `NetworkPolicy` that will be evaluated before the BANP.
```

Comment:
I structured this multi-tenancy based on your blogpost which is where I took the examples from for ANP Pass action and BANP Deny. When you combine those examples with this, you get the multi-tenancy policy I'm discussing here and you were discussing in your blog post. @tssurya. Which is to say, this is not clear and needs to come out for not till 4.16, I think.

Reply:
but when I took an initial review stab at this it was totally incomplete, I think the pieces are now complete based on the blogpost and make more sense. They can stay and are pretty useful and should tie up loose ends with the diagram from top. Having said that I can't review these bits because I wrote them so refrainng from doing so..